scrc/dataset_creation/criticality_dataset_creator.py

```diff
@@ -69,40 +76,42 @@ def extract_bge_references(self):
         if not bge_references_file_path.exists():
             raise Exception("bge references need to be extracted first. Run bge_reference_extractor.")
         df = pd.DataFrame({'bge_file_number_long': [], 'bge_file_number_short': [], 'bger_reference': [], 'year': [],
-                           'bge_chamber': []})
+                           'bge_chamber': [], 'bger_chamber': [], 'legal_area': []})
         with bge_references_file_path.open("r") as f:
             for line in f:
                 (bge_file_number_long, bger_reference) = line.split()
+                bger_chamber = bger_reference.split('_', 2)[0]
+                legal_area = self.get_legal_area_bger(bger_chamber)
                 year = int(bge_file_number_long.split('-', 5)[1]) + 1874
                 bge_chamber = bge_file_number_long.split('_', 5)[2]
                 bge_file_number_short = bge_file_number_long.split('_')[3]
-                s_row = pd.Series([bge_file_number_long, bge_file_number_short, bger_reference, year, bge_chamber],
+                s_row = pd.Series([bge_file_number_long, bge_file_number_short, bger_reference, year, bge_chamber, bger_chamber, legal_area],
                                   index=df.columns)
                 df = df.append(s_row, ignore_index=True)
         self.logger.info(
             f"References_df: There are {len(df.index)} entries with {len(df.bge_file_number_long.unique())} unique bge_filenumbers and {len(df.bger_reference.unique())} unique bger_references.")
+        folder = self.create_dir(self.get_dataset_folder(), f'reports/references')
+        report_creator = ReportCreator(folder, self.debug)
+        report_creator.report_references(df)
         return df
 
-    def extend_file_number_list(self, file_number_list):
-        """
-        extend a given list of bger_file_numbers with unknown syntax to a list which contains all possible syntax
-        :param file_number_list:    list of bger_file_numbers with unknown syntax
-        :return:                    list with all 3 syntaxes of each bger_file_number found in given list
-        """
-        # this extension is needed because file_numbers have no common syntax.
-        bge_list = []
-        for i in file_number_list:
-            i = i.replace(' ', '_')
-            i = i.replace('.', '_')
-            (chamber, text) = i.split("_")
-            all = [f"{chamber} {text}", f"{chamber}.{text}", f"{chamber}_{text}"]
-            bge_list.extend(all)
-        return bge_list
+    def get_legal_area_bger(self, chamber_number):
+        switch = {
+            1: 'public_law',
+            2: 'public_law',
+            4: 'civil_law',
+            5: 'civil_law',
+            6: 'penal_law',
+            8: 'social_law',
+            9: 'social_law'
+        }
```

Comment:
this looks like something that should be refactored out to a more general class and be used in the dataset_creator.

Reply:
This will only be used for bger references in bge. I can move it to sql_utils, there are already similar methods.